examples/graphs/node2vec_movielens.py

```diff
@@ -0,0 +1,584 @@
+
+[node2vec](https://arxiv.org/abs/1607.00653) is a simple, yet scalable and effective
+technique for learning low-dimensional embeddings for nodes in a graph by optimizing
+a neighborhood preserving objective. The aim is to learn similar embeddings for
+neighboring nodes, with respect to the graph structure.
+
+Given your data items structured as a graph (where the items are represented as
+nodes and the relationship between items are represented as edges),
+node2vec works as follows:
+
+1. Generate item sequences using (biased) random walk.
+2. Create positive and negative training examples from these sequences.
+3. Train a [word2vec](https://www.tensorflow.org/tutorials/text/word2vec) model
+(skip-gram) to learn embeddings for the items.
+
+In this example, we demonstrate the node2vec technique on the [small version of the Movielens dataset](https://files.grouplens.org/datasets/movielens/ml-latest-small-README.html)
+to learn movie embeddings. Such a dataset can be represented as a graph by treating
+the movies as nodes, and creating edges between movies that have similar ratings
+by the users. The learnt movie embeddings can be used for tasks such as movie recommendation,
+or movie genres prediction.
+
+This example requires `networkx` package, which can be installed using the following command:
+
+```
+pip install networkx
+````
+"""
+
+"""
+## Setup
+"""
+
+import os
+from collections import defaultdict
+import math
+import networkx as nx
+import random
+from tqdm import tqdm
+from zipfile import ZipFile
+from urllib.request import urlretrieve
+import numpy as np
+import pandas as pd
+import tensorflow as tf
+from tensorflow import keras
+from tensorflow.keras import layers
+import matplotlib.pyplot as plt
+
+"""
+## Download and Prepare the Data
+
+The small version of the Movielens dataset includes around 100 thousand ratings
+from 610 users on 9742 movies.
+
+First, let's download the movielens data. The downloaded folder will contain
+three data files: `users.csv`, `movies.csv`, and `ratings.csv`. In this example,
+we will only need the `movies.dat`, and `ratings.dat` data files.
+"""
+
+urlretrieve(
+    "http://files.grouplens.org/datasets/movielens/ml-latest-small.zip", "movielens.zip"
+)
+ZipFile("movielens.zip", "r").extractall()
+
+"""
+Then, we load the data into pandas DataFrames and perform some basic processing.
+"""
+
+# Load movies to a DataFrame.
+movies = pd.read_csv("ml-latest-small/movies.csv")
+# Make the movieId string.
+movies["movieId"] = movies["movieId"].apply(lambda x: f"movie_{x}")
+
+# Load ratings to a DataFrame.
+ratings = pd.read_csv("ml-latest-small/ratings.csv")
+# # Convert rating to float
+ratings["rating"] = ratings["rating"].apply(lambda x: float(x))
+# # Make the movie_id string.
+ratings["movieId"] = ratings["movieId"].apply(lambda x: f"movie_{x}")
+
+print("Movies data shape:", movies.shape)
+print("Ratings data shape:", ratings.shape)
+
+"""
+Let's show a sample instance of the `ratings` DataFrame.
```

Comment:
Suggestion:

- "Show" -> "Inspect" (a part of a Pandas DataFrame)

```suggestion
Let's inspect a sample instance of the `ratings` DataFrame.
```